infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/CheInstallationLocation.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.environment;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+
+/**
+ * This class checks the KUBERNETES_NAMESPACE and POD_NAMESPACE environment variables to determine
+ * what namespace Che is installed in. Users should use this class to retrieve the installation
+ * namespace name.
+ *
+ * @author Tom George
+ */
+@Singleton
+public class CheInstallationLocation {
```

Comment:
This would be better included in this changeset as a `git mv` rather than deleting the earlier file.

Reply:
I don't think it would help. imho github or some git visualiser (git log, gitk, etc) are doing some magic to detect moved files, but this one probably changed too much to detect. I don't think git engine itself track this. But I'm not sure, if you know how to do it now retrospectively, I can try.